pkg/services/pluginsintegration/plugininstaller/service.go

```diff
@@ -2,31 +2,76 @@ package plugininstaller
 
 import (
 	"context"
+	"errors"
+	"runtime"
+	"strings"
 
 	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
-	features featuremgmt.FeatureToggles
-	log      log.Logger
+	cfg             *setting.Cfg
+	features        featuremgmt.FeatureToggles
+	log             log.Logger
+	pluginInstaller plugins.Installer
+	pluginStore     pluginstore.Store
 }
 
-func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles) *Service {
+func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, pluginStore pluginstore.Store, pluginInstaller plugins.Installer) *Service {
 	s := &Service{
-		features: features,
-		log:      log.New("plugin.installer"),
+		features:        features,
+		log:             log.New("plugin.backgroundinstaller"),
+		cfg:             cfg,
+		pluginInstaller: pluginInstaller,
+		pluginStore:     pluginStore,
 	}
 	return s
 }
 
 // IsDisabled disables background installation of plugins.
 func (s *Service) IsDisabled() bool {
-	return !s.features.IsEnabled(context.Background(), featuremgmt.FlagBackgroundPluginInstaller)
+	return !s.features.IsEnabled(context.Background(), featuremgmt.FlagBackgroundPluginInstaller) ||
+		len(s.cfg.InstallPlugins) == 0
 }
 
 func (s *Service) Run(ctx context.Context) error {
-	s.log.Debug("PluginInstaller.Run not implemented")
+	compatOpts := plugins.NewCompatOpts(s.cfg.BuildVersion, runtime.GOOS, runtime.GOARCH)
+
+	for _, pluginIDRaw := range s.cfg.InstallPlugins {
+		parsed := strings.Split(pluginIDRaw, "@")
+		pluginID := parsed[0]
+		version := ""
+		if len(parsed) == 2 {
+			version = parsed[1]
+		}
+
+		// Check if the plugin is already installed
+		p, exists := s.pluginStore.Plugin(ctx, pluginID)
+		if exists {
+			// If it's installed, check if we are looking for a specific version
+			if version == "" || p.Info.Version == version {
+				s.log.Debug("Plugin already installed", "pluginID", pluginID, "version", version)
```

Comment:
We are trying to stick to standard camelCase when it comes to the logs 🙈 Confusing I know..

```suggestion
				s.log.Debug("Plugin already installed", "pluginId", pluginID, "version", version)
```